internal/cli/bud/bud.go

```diff
@@ -0,0 +1,178 @@
+package bud
+
+import (
+	"context"
+	"io"
+	"os"
+	"os/exec"
+
+	"github.com/livebud/bud/internal/current"
+	"github.com/livebud/bud/internal/pubsub"
+
+	"github.com/livebud/bud/framework"
+	"github.com/livebud/bud/framework/app"
+	"github.com/livebud/bud/framework/controller"
+	"github.com/livebud/bud/framework/public"
+	"github.com/livebud/bud/framework/transform/transformrt"
+	"github.com/livebud/bud/framework/view"
+	"github.com/livebud/bud/framework/view/dom"
+	"github.com/livebud/bud/framework/view/ssr"
+	"github.com/livebud/bud/framework/web"
+	"github.com/livebud/bud/package/commander"
+	"github.com/livebud/bud/package/di"
+	"github.com/livebud/bud/package/gomod"
+	v8 "github.com/livebud/bud/package/js/v8"
+	"github.com/livebud/bud/package/log"
+	"github.com/livebud/bud/package/log/console"
+	"github.com/livebud/bud/package/log/filter"
+	"github.com/livebud/bud/package/overlay"
+	"github.com/livebud/bud/package/parser"
+	"github.com/livebud/bud/package/socket"
+	"github.com/livebud/bud/package/svelte"
+)
+
+// Input contains the configuration that gets passed into the commands
+type Input struct {
+	Stdin  io.Reader
+	Stdout io.Writer
+	Stderr io.Writer
+	Env    []string
+
+	// Currently passed in only for testing
+	Dir   string          // Can be empty
+	BudLn socket.Listener // Can be nil
+	WebLn socket.Listener // Can be nil
+	Bus   pubsub.Client   // Can be nil
+}
+
+func New(in *Input) *Command {
+	return &Command{in: in}
+}
+
+type Command struct {
+	in   *Input
+	Dir  string
+	Log  string
+	Args []string
+	Help bool
+}
+
+// Run a custom command
+// TODO: finish supporting custom commands
+// 1. Compile
+//   a. Generate generator (later!)
+//   	 i. Generate bud/internal/generator
+//     ii. Build bud/generator
+//     iii. Run bud/generator
+//   b. Generate custom command
+//     i. Generate bud/internal/command/${name}/
+//     ii. Build bud/command/${name}
+// 2. Run bud/command/${name}
+func (c *Command) Run(ctx context.Context) error {
+	return commander.Usage()
+}
+
+// Module finds the go.mod file for the application
+func Module(dir string) (*gomod.Module, error) {
```

Comment:
internal/cli/bud contains some shared configuration, the root command and some helper functions that are used by the rest of the CLI